web_external/views/body/SubmissionView.js

```diff
@@ -60,32 +79,33 @@ covalic.views.SubmissionView = covalic.View.extend({
         // Must unbind previous fetch handlers to avoid infinite callbacks
         this.submission.off('g:fetched');
 
-        this.phase = new covalic.models.PhaseModel({
+        this.phase = new PhaseModel({
             _id: this.submission.get('phaseId')
         }).on('g:fetched', function () {
             this.render();
 
             if (!this.submission.get('score')) {
-                girder.eventStream.on('g:event.job_status', this._statusHandler, this);
-                girder.eventStream.on('g:event.progress', this._progressHandler, this);
+                eventStream.on('g:event.job_status', this._statusHandler, this);
+                eventStream.on('g:event.progress', this._progressHandler, this);
 
-                if (girder.currentUser && (
-                        girder.currentUser.get('_id') === this.submission.get('creatorId') ||
-                        girder.currentUser.get('admin'))) {
-                    this.job = new girder.models.JobModel({
+                var currentUser = getCurrentUser();
+                if (currentUser && (currentUser.id === this.submission.get('creatorId') ||
+                                    currentUser.get('admin'))) {
+                    this.job = new JobModel({
                         _id: this.submission.get('jobId')
                     }).on('g:fetched', function () {
-                        if (this.job.get('status') === girder.jobs_JobStatus.ERROR) {
+                        if (this.job.get('status') === JobStatus.ERROR) {
```

Comment:
The `fetch()` should not be chained anymore: 
Uncaught TypeError: this.job.get is not a function`
Perhaps there are more problems like this one?